test/contrib/rails/errors_test.rb

```diff
@@ -54,6 +54,8 @@ class TracingControllerTest < ActionController::TestCase
   end
 
   test 'missing rendering should close the template Span' do
+    skip 'Recent versions use events, and cannot suffer from this issue' if Rails.version >= '4.0.0'
```

Comment:
For versions of Rails that do not support the event-based integration, we start a span for a `rails.render_template` even if that template does not exist in the application.
Rails then fails to load it and errs out.
This leaves us with a `rails.render_template` that we have to finish somehow, so we make it as an error span.

Ideally, this span wouldn't exist, as the application actually never got to start rendering something that doesn't exist. Besides, we already have error information in the parent span, so no additional information is conveyed with that "fake" span.

In the even-based implementation, that's the case: we do no have a span for a template that has never existed in first place. Rails won't even fire a "start rendering" event for it, as it gives up before it start to render.
This makes this specific test no relevant to the event-based implementation, as its main goal is to check that the nested missing `rails.render_template` span was successfully closed.

Reply:
Okay, perhaps we should consider adding a comment that reminds us that these parts of code should be removed when we drop 3.2 support.

Reply:
By having the line `skip '...' if Rails.version >= '4.0.0'` was hoping that this would be sufficient for us to notice and remove it when dropping support for the last version of Ruby `3.X`.